driver/userspace.c

```diff
@@ -460,6 +460,33 @@ WnbdDeleteConnection(PWNBD_EXTENSION DeviceExtension,
     return STATUS_SUCCESS;
 }
 
+_Use_decl_annotations_
+NTSTATUS
+WnbdSetDiskSize(PWNBD_EXTENSION DeviceExtension, WNBD_CONNECTION_ID ConnectionId, UINT64 BlockCount) {
+    ASSERT(DeviceExtension);
+    ASSERT(ConnectionId);
+
+    PWNBD_DISK_DEVICE Device = WnbdFindDeviceByConnId(DeviceExtension, ConnectionId, TRUE);
+
+    if (!Device) {
+        WNBD_LOG_ERROR("Could not find the device to resize.");
+        return STATUS_OBJECT_NAME_NOT_FOUND;
+    }
+
+    Device->Properties.BlockCount = BlockCount;
+    WNBD_LOG_INFO("Resized disk: %s. "
```

Comment:
you'll have to move the log message above the assignment, otherwise you override the old value before logging it.